### ISSUE TITLE
Reports

### ISSUE THREAD

**Issue by lkwirica** [intrahealth/emNutt#21]
Report required on messages sent, time and recipient for proper audit trail. Reports required should include:

Sent Messages analysis by:
•	County: No. of messages being sent out to each county
•	Cadre: Numbers reached per SMS in each cadre
•	Date: A list of all messages sent and which day the message was sent
•	Running total: Total Number of messages sent out to HWs
•	Failed: The list of HWs to whom messages failed to be sent. (failed list)

Feedback analysis by:
•	Gender-How many male/female responses
•	Cadre
•	County-How many responses are we receiving per county
•	Total Responses- by Date/week/month/day of week

**Comment by cloehr651** [comment]
To make the best use of the ticket system we urge you to only list one request per "issue". I will split this up as well as ensure there isn't overlap with the existing issues related to reports.

**Comment by cloehr651** [comment]
@lkwirica, To make the best use of the ticket system we urge you to only list one request per "issue". Once I get clarification I will split this up or merge it with the existing issues related to reports. 

Are you envisioning a single report for each broadcast message or message flow, or a report that presents all of that information for all broadcast message or message flow? It seems like County and Cadre (from the Sent Message analysis request) would be a lot of detail to include in a report that lists all sent broadcast messages/flows so would fit better in a report with results related to a specific broadcast message/flow. Please confirm if this seems like it would fit your needs.

A similar question about the feedback analysis - are you interested in a report that provides this info for each sent broadcast messages/message flows, or for all?

**Comment by cloehr651** [comment]
@lkwirica, can you take a look at the reports I have drafted in [this document](https://drive.google.com/file/d/14K9HAa_9oq74bzkJcX3mQBrosl3GAxw-/view?usp=sharing). With Kibana it will be easy for each country to choose which filters they want to have available so I have just prepared the standard report without customizations. Please take a look and see if you think this will cover all of your needs/desires - I'm happy to make any changes if something is missing or if you have suggestions on other ways to present the information.

**Comment by lkwirica** [comment]
@cloehr651 please see my feedback on the mock up shared.

For Purposes of simplicity, it’s best to separate the Inbox/Outbox/ Sent/ Failed folders from the Reports Centre. i.e. maintain the folders with a list of messages sent/received, and have a separate feature for reports.

**Responses or Inbox** 
On the inbox messages, is the received message column necessary? If the contact did not receive the message, we cannot have a corresponding question/message. 
Also, I believe that the flow name is pulled from Rapidpro. Where is the message name stored for messages sent directly via mHero i.e. no workflow is used to send the message. 
Equally, how can we assign an inbox message to a message sent if it is not part of predefined responses to a workflow? 
The ability to export filtered data (to .xls/ csv) should be introduced. 

**Sent and Outbox**

Introduce the ability to export filtered data to .xlsx/.csv formats

**Status By**
Since the sent/completed messages are already captured in Sent and Outbox, I believe it is easier to simply maintain a record of failed messages.
The status (completed, pending etc) can be included as columns in the Sent, Outbox.
Date Started/Date Ended needs clarification. For purposes of Failed messages, only date sent (date the message was sent and failed to send) is necessary.
The key fields in Failed Box should be:
•	Names
•	Phone Number
•	Date 
The ability to filter by  date and export to excel formats should be included. This will make it easy to investigate the cause of failure, and correct/update any issues.

**Reports Centre**
Sent Messages analysis by:
•	County: No. of messages being sent out to each county
•	Cadre: Numbers reached per SMS in each cadre
•	Date: A list of all messages sent and which day the message was sent
•	Running total: Total Number of messages sent out to HWs
•	Failed: The list of HWs to whom messages failed to be sent. (failed list)

Feedback analysis by:
•	Gender-How many male/female responses
•	Cadre
•	County-How many responses are we receiving per county
Total Responses- by Date/week/month/day of week

I'll share a suggestion of how the reports can be displayed

**Comment by cloehr651** [comment]
Thank you very much, @lkwirica, for your detailed feedback!

We will discuss separating the inbox/outbox/sent/failed from the reports section. Exporting data is something I have already requested so Ally is aware of this. I also suggested adding the option to filter, per your feedback shared at the end of last year.

Regarding the inbox, thank you for pointing out that the "received" column doesn't make sense there and the issue with the message name if sending from within mHero. For incoming messages that aren't in response to a flow, we could either pull the name of the most recent message/flow or else it could be blank. I think having the message name is helpful for giving the messages some context.

Regarding the status report, thank you also for letting us know that this report seems like a duplicate of the sent/outbox. I was trying to incorporate all of the feedback you shared at the end of last year but can see now how it overlaps. I recommend that we include the flow/message name as well as the contact name, again so that there is context.

Lastly, about the display for the reports center, we have a more detailed and visually appealing proposal but we are waiting on getting a consultant or a new team member who can develop it. Please do share your suggestion and then we can harmonize your idea with ours.

**Comment by lkwirica** [comment]
@ashaban the Message sending summary is a great report. Very precise and brief. 

I suggest the  message sending breakdown build on the Message sending summary i.e. Once I click on the Sent Number, I am able see /download in .csv, a list of the sent, and once I click on the failed, I'm able to do the same.

Also, for each of the reports, ability to export is very key. 

The same goes for Flow Run Summary/Breakdown

**Comment by cloehr651** [comment]
@lkwirica can you review the new reports Ally added (Flow Run Breakdown and Message Sending Breakdown) and confirm whether these reports address all of the requests in this issue? If there are other requests/issues remaining I suggest creating new "issues" for each so that it is easier to understand the request and track the status.

**Comment by cloehr651** [comment]
@lkwirica can you please review the reports and confirm whether these reports address all of the requests in this issue?

**Comment by lkwirica** [comment]
The reports on mHero address Messages sent out. The various folders/ reports for received messages is pending.

**Comment by njugunask** [comment]
Sent messages Analysis report – **Done**

**Below still pending**

Feedback analysis – Two items to be implemented (display: Response by Gender, Total responses by date/week/month/day of the week)

**Comment by njugunask** [comment]
****Feedback analysis by**:-**
- Cadre & County > this is available

**Pending:** 
- analysis by Gender
- Total Responses by:- Date, Week, Month & Day of the week

A suggestion, to enable seeing reports or messages sent in a particular, **Day** or **Month**, unmerge the column **Date** & **Time** so that they are reported separately, then it is possible to filter out messages sent out in a particular day or month regardless of the time within that day it was sent.

**Comment by cloehr651** [comment]
@njugunask there are several new issues here. For the sake of clarity and tracking, please create new separate issues for each request.

**Comment by njugunask** [comment]
@cloehr651, the tasks are one and the same thing raised by @lkwirica. Some of them have already been fixed so following up with the ones remaining

**Comment by lkwirica** [comment]
> ****Feedback analysis by**:-**
> 
> * Cadre & County > this is available
> 
> **Pending:**
> 
> * analysis by Gender
> * Total Responses by:- Date, Week, Month & Day of the week
> 
> A suggestion, to enable seeing reports or messages sent in a particular, **Day** or **Month**, unmerge the column **Date** & **Time** so that they are reported separately, then it is possible to filter out messages sent out in a particular day or month regardless of the time within that day it was sent.

@njugunask Feedback analysis by cadre/county is not available. What is available is an analysis for sent messages. Not received.

**Comment by cloehr651** [comment]
@njugunask and @lkwirica , even if there is only one issue here, my initial comment on this issue when it was first created is that it's not clear what the issue is just by looking at the title and since there were so many issues included in the initial request it is hard to follow what has been done or not. **Please create another issue with a clear title and description of the outstanding issue.**

**Comment by ashaban** [comment]
I agree with @cloehr651

**Comment by njugunask** [comment]
@lkwirica, kindly revisit this issue and clarify

**Comment by ashaban** [comment]
@lkwirica  I have added summary by gender as requested
The inbox report is available as well and it is filterable by gender, county, facility, cadre etc
Let me know if there is anything outstanding on this issue and if possible create a new issue so that we track them differently instead of combining multiple issues on just on issue thread.
